server.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const express = require('express');
+const exphbs = require('express-handlebars');
+const methodOverride = require('method-override');
+const session = require('express-session');
+const bodyParser = require('body-parser');
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+const bcrypt = require('bcryptjs');
+const redis = require('connect-redis')(session);
+
+const routesListings = require('./routes/listings');
+const routesGallery = require('./routes/gallery');
+const routesLogin = require('./routes/login');
+const routesRegister = require('./routes/register');
+const routesLogout = require('./routes/logout');
+
+const User = require('./database/models/User');
+const PORT = process.env.PORT || 8080;
+const ENV = process.env.development || 'development';
+const SESSION_SECRET = process.env.SESSION_SECRET || 'keyboard cat';
+
+const app = express();
+
+app.use(methodOverride('_method'));
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+app.use(session({
+  store: new redis({ url: 'redis://redis-server:6379', logErrors: true }),
+  secret: SESSION_SECRET,
+  resave: false,
+  saveUninitialized: false,
+  cookie: { secure: ENV === 'production' }
+}));
+
+app.engine('.hbs', exphbs({
+  extname: '.hbs',
+  defaultLayout: 'main.hbs'
+}));
+app.set('view engine', '.hbs');
+
+app.use(express.static('public'));
+app.use('/gallery', express.static('public'));
```

Comment:
This doesn't make sense since you're trying to make everything in the /gallery route just serve everything in your public dir? I would also remove this since you seem to override it later in this file.